### ISSUE TITLE
Running on 7.31 or 7.02 system fails

### ISSUE THREAD

**Issue by kuntergunt** [larshp/abapGit#3714]
To me it sems like abapGit is no more 7.02 compliant. I also added it to a 7.31 system and got even there some errors. The error says that the Message-Id is missing in the program declaration. But adding it there it just came up with more similar errors on all message statements.

Tested with abapGit version 1.98.0.
Going back to version 1.89.0 solved the problems.

tested on this system:
SAP_BASIS       731    0027    SAPKB73127    SAP Basiskomponente
SAP_ABA	        731    0027    SAPKA73127    Anwendungsübergreifende Komponente

**Comment by larshp** [comment]
thanks, can you help adding a screenshot of the exact error message?

**Comment by kuntergunt** [comment]
![image](https://user-images.githubusercontent.com/11312349/89525583-b5c11000-d7e6-11ea-8397-82e4bc3e7823.png)

**Comment by larshp** [comment]
and please also add what is in line 32818, the program is updated often, so the latest might not be identical to what this error message refers to

**Comment by kuntergunt** [comment]
It would be ok for me if the support for systems < 7.50 is being skipped. But a version (even non maintained) that is compatible with these old versions would be nice for download.

**Comment by kuntergunt** [comment]
![image](https://user-images.githubusercontent.com/11312349/89525872-2a944a00-d7e7-11ea-937b-a4cf8d1bfe18.png)

**Comment by kuntergunt** [comment]
Looks like this is all related to message-id and message statements. I started editing but after I saw how many changes were needed I skipped and went for an older version.

**Comment by larshp** [comment]
this was changed 6 days ago, https://github.com/larshp/abapGit/blame/master/src/ui/zcl_abapgit_popups.clas.abap

how many syntax errors are there? from the above screenshot it looks like there is just this one?

try the following instead of line 32818,

```abap
DATA lv_message TYPE string.
lv_message = |Branch switched from { zcl_abapgit_git_branch_list=>get_display_name( iv_default_branch ) } to { zcl_abapgit_git_branch_list=>get_display_name( rs_branch-name ) }|.
MESSAGE lv_message TYPE 'S'.
```

**Comment by kuntergunt** [comment]
This change fixed the compilation errors on the version 7.31 system!
Unfortunately the 7.02 system has more issues. Though I don't really need it on this one, here are the errors from the 7.02 system:

![image](https://user-images.githubusercontent.com/11312349/89532672-76002580-d7f2-11ea-8e5b-c0f7e012121e.png)

**Comment by larshp** [comment]
hmm, that looks strange, pragmas are supported in 702 as far as I know, https://blogs.sap.com/2010/12/15/abap-feature-set-backport-7x-to-702-pragmas/, https://github.com/sapmentors/abap2xlsx/issues/331

**Comment by ed-holland** [comment]
You have to be careful. There are pragmas in 702 but only those are accepted.
Remember that i put the pragmas as comments instead so that they could be used and not fail on older releases.

**Comment by FreHu** [comment]
#3555 we could just remove them

**Comment by larshp** [comment]
yea, but wonder if there are more pragmas causing problems, or if its all pragmas

**Comment by sbcgua** [comment]
from what I remember (read it somewhere) that all pragmas ` code ##PRAGMA.` may be causing issues on 702. 702 supports only ` code. " PRAGMA`

**Comment by larshp** [comment]
according to the official documentation, https://help.sap.com/doc/abapdocu_752_index_htm/7.52/en-US/abennews-71-pragmas.htm, pragmas are supported in 702

**Comment by larshp** [comment]
@kuntergunt can you help adding a screenshot of the exact SAP_BASIS version? kernel version etc.

**Comment by mbtools** [comment]
I fired up my 702 SP 21 (Kernel 722_EXT_REL patch 900) and installed the latest ZABAPGIT_STANDALONE.

There is indeed the error with the message statement reported above. However, after the fix provided by Lars, the program activates and runs successfully. 

![image](https://user-images.githubusercontent.com/59966492/89574860-d2674300-d7fa-11ea-8715-d82e71428cfe.png)

There are a few warnings but these can be ignored. 

![image](https://user-images.githubusercontent.com/59966492/89575145-56b9c600-d7fb-11ea-8c92-2d9aa3e20475.png)

**Comment by mbtools** [comment]
Gunter, 

the following code should be on line 63 but it shows on line 68 in your screenshot above.

    rv_text = 'Error in two factor authentication.' ##NO_TEXT.

Pls double check that you are using the most recent code in 702.

**Comment by kuntergunt** [comment]
> @kuntergunt can you help adding a screenshot of the exact SAP_BASIS version? kernel version etc.
SAP_ABA   700   0021
SAP_BASIS   700   0021

Kernel-Release    722
Compilation       Linux GNU SLES-9 x
Patch-Level       201
ABAP-Load         1780
CUA-Load          41
Mode              opt

**Comment by kuntergunt** [comment]
> Gunter,
> 
> the following code should be on line 63 but it shows on line 68 in your screenshot above.
> 
> ```
> rv_text = 'Error in two factor authentication.' ##NO_TEXT.
> ```
> 
> Pls double check that you are using the most recent code in 702.

I have tested with different versions, obviously the one I took the screenshot from was not the latest. I installed the latest version nor and got errors:
Programm ZABAPGIT_FULL	63
Fehlerhafter arithmetischer oder Bitausdruck: Statt "##NO_TEXT" wurde
ein Operator (+, -, *, /, ... bzw. BIT-AND, BIT-XOR, BIT-OR) erwartet.	

So I removed all the ##NO_TEXT pragmas. Then I got:
Programm ZABAPGIT_FULL	336	
Nach "CONSTRUCTOR" wurde "." oder "BY KERNEL MODULE ..." erwartet.	

Removing this pragma gave me even more errors:	
Programm ZABAPGIT_FULL	454	
Feld "|ZCX_ABAPGIT_EXCEPTION*|" unbekannt. Es ist weder in einer der		
angegebenen Tabellen enthalten noch durch eine "DATA"-Anweisung		
definiert.		
Programm ZABAPGIT_FULL	1.471
Nach "KEY" wurde "," oder "INITIAL SIZE ... ." erwartet.
